src/App.js

```diff
@@ -14,11 +15,33 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    goodsList: '',
+  }
+
+  render() {
+    return (
+      <div className="App">
+        <h1>Goods</h1>
+        {this.state.goodsList}
+        {this.state.goodsList === ''
+          && (
+          <button
+            type="button"
+            onClick={() => {
+              this.setState({
+                goodsList: <GoodsList goodsList={goodsFromServer} />,
```

Comment:
all that you need to save in state that's a boolean value and render GoodsList depend on that value